server/service/src/sync/translations/master_list.rs

```diff
@@ -38,6 +42,21 @@ impl SyncTranslation for MasterListTranslation {
             PullUpsertRecord::MasterList(result),
         )))
     }
+
+    fn try_translate_pull_delete(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        let result = match_pull_table(sync_record).then(|| {
+            IntegrationRecords::from_delete(
```

Comment:
just use an `if` instead of `.then`

Reply:
Why ?

Reply:
Thanks for review, but this is not helpful, if there is an issue with what you see please explain why. Similar to the comment above.

Reply:
just a suggestion to use a more straight forward `if`

Reply:
maybe i am not comparing apples with apples here

```
let new_option = option.map(|o| o.inner_value)
```

vs

```
let new_option = if let Some(value) = new_option {
  Some(value.inner)
}
```

Yep 'then' is new syntax, but once you know, you know 🤷